src/core/thread/mle.cpp

```diff
@@ -3730,33 +3753,13 @@ otError Mle::HandleDiscoveryResponse(const Message &aMessage, const Ip6::Message
             aMessage.Read(offset, sizeof(steeringData), &steeringData);
             VerifyOrExit(steeringData.IsValid(), error = OT_ERROR_PARSE);
 
-            // Pass up MLE discovery responses only if the steering data is set to all 0xFFs,
-            // or if it matches the factory set EUI64.
-            if (mEnableEui64Filtering)
+            if (mDiscoverEnableFiltering)
             {
-                Mac::ExtAddress extaddr;
-                Crc16           ccitt(Crc16::kCcitt);
-                Crc16           ansi(Crc16::kAnsi);
-
-                otPlatRadioGetIeeeEui64(&GetInstance(), extaddr.m8);
-
-                MeshCoP::ComputeJoinerId(extaddr, extaddr);
-
-                // Compute bloom filter
-                for (size_t i = 0; i < sizeof(extaddr.m8); i++)
-                {
-                    ccitt.Update(extaddr.m8[i]);
-                    ansi.Update(extaddr.m8[i]);
-                }
-
-                // Drop responses that don't match the bloom filter
-                if (!steeringData.GetBit(ccitt.Get() % steeringData.GetNumBits()) ||
-                    !steeringData.GetBit(ansi.Get() % steeringData.GetNumBits()))
-                {
-                    ExitNow(error = OT_ERROR_NONE);
-                }
+                VerifyOrExit((steeringData.GetBit(mDiscoverCcittIndex % steeringData.GetNumBits()) &&
```

Comment:
We don't want to return an error here since we log the error at `exit` label (at warning level). The Discovery response itself is not bad or incorrect. We just filtered it (we are not looking for this network). 

We can log the fact that steering data did include this device and we are filtering the scan result, but not sure how useful it would be (we anyways log the results that pass filer in `Joiner` itself). 
IIRC, we used to log a similar log line when we filtered non-joinable networks and it misled a lot folks looking at the logs during pairing thinking there was an error with scan operation (specially in test areas where they are many networks and scan result can be large).